rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java

```diff
@@ -144,13 +147,110 @@ void testGetOldFederation(
 
     @Test
     void getOldFederation_previouslySetToNull_returnsNull() {
-        StorageAccessor storageAccessor = new InMemoryStorage();
-        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(storageAccessor);
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(null);
```

Comment:
The `null` part is to show in the test that it will not be used. It is not expected to be used. I think for test it is reasonable to have it this way. What do you think? This is mainly because we are avoiding mocks. With a mock, I would simply assert that certain mocked function is not called, etc. But with an actual StorageAccessor, there's no way to ensure it was not accessed, unless I don't set it, and if it were to be used when we are not expecting it, then the test would fail.

Reply:
I don't like this approach. It's basically setting a null value to then show that if a NPE is not thrown then the value was not used. Not very nice in my opinion.

You can always check the storage before/after to make sure it didn't change if that's what we need to assert

Reply:
Well, I kind of agree with Marcos; I don't like the NPE idea. What about using a mock and verifying that the mock method is a `never` call?

Reply:
Updated.

Reply:
i doesn't appear updated for me, still `null`

Reply:
Updated.